entrypoint.py

```diff
@@ -22,33 +27,35 @@ def _print_cmds(self):
         for i in self.cmds:
             print(i)
 
-    def _get_image(self, img_hash):
-        imgs = self.cli.images()
-        for i in imgs:
-            if img_hash in i['Id']:
-                self.img = i
-                return
-        raise ImageNotFound("Image {} not found\n".format(img_hash))
+    def _get_image(self, image_id):
+        try:
+            return self.client.images.get(image_id)
+        except:
+            raise ImageNotFound("Image {} not found\n".format(image_id))
 
     def _insert_step(self, step):
+        # ignore the end "# buildkit" comment
+        step = re.sub("\s*#.+$", "", step)
         if "#(nop)" in step:
             to_add = step.split("#(nop) ")[1]
         else:
-            to_add = ("RUN {}".format(step))
+            # step may contains "/bin/sh -c ", just ignore it
+            to_add = "{}".format(step.replace("/bin/sh -c ", ""))
```

Comment:
Maybe this should be changed in another PR ? The results with the current docker image and one built from your branch are quite different:

With the current image:

```
❯ docker run --rm -v '/var/run/docker.sock:/var/run/docker.sock' lukapeschke/dfa 21ab84251c7b
FROM lukapeschke/dfa:latest
ADD file:093f0723fa46f6cdbd6f7bd146448bb70ecce54254c35701feeceb956414622f in /
CMD ["/bin/sh"]
RUN /bin/sh -c apk add --update python3 wget      \
    && wget -O - --no-check-certificate https://bootstrap.pypa.io/get-pip.py | python3      \
    && apk del wget      \
    && pip3 install -U docker-py      \
    && yes | pip3 uninstall pip
COPY file:df2a17029ab5f655279e92b51f29046f182eb0a4a5457ad06b2338387bcbbd20 in /root
ENTRYPOINT ["/root/entrypoint.py"]
```

With the image build from your PR:

```
❯ docker run --rm -v '/var/run/docker.sock:/var/run/docker.sock' updated-image 21ab84251c7b
FROM lukapeschke/dfa:latest
/bin/sh -c
/bin/sh -c
apk add --update python3 wget      \
    && wget -O - --no-check-certificate https://bootstrap.pypa.io/get-pip.py | python3      \
    && apk del wget      \
    && pip3 install -U docker-py      \
    && yes | pip3 uninstall pip
/bin/sh -c
/bin/sh -c
```

Reply:
I updated the code 😄 It should fixed in the following commits.